software-engineering-policies/CloudDevelopment/CloudDevelopmentPolicy.md

```diff
@@ -0,0 +1,329 @@
+
+#### Modules
+
+The DDC team manages a collection of centralised Terraform modules found here: https://github.com/orgs/UKHO/teams/tfmodules/repositories
+These should be used whenever possible.
+
+#### IaC Scanning (Trivy)
+
+Trivy is UKHO's Chosen IaC Scanning tool. All Infrastructure pipelines should include a Trivy Scan stage before deployment that fails the build if misconfigurations or vulnerabilities are detected.
+A Trivy Stage is included in the previously mentioned Terraform deployment template: https://github.com/UKHO/Terraform-Deployment-Template/blob/5ca2e1c18618dad0df84ef8c08aea87a64c3b94e/azure-pipelines.yml#L37
+
+### Secrets Management
+
+#### Centralized Secrets Management
+
+All secrets, credentials, and sensitive data used by applications must be stored in Azure Key Vault.
+   
+#### Access Control
+
+Access to Key Vault must be restricted using Azure RBAC and Key Vault Access Policies, with access granted based on the principle of least privilege.
+
+#### Network Security
+
+Enable Private Endpoints or Service Endpoints for Key Vault to restrict access to trusted networks.
+
+#### Secret Rotation and Expiration
+
+Secrets must be rotated on a regular basis, with automation for renewal and alerting in place for any approaching expiration.
+
+#### Audit and Monitoring
+
+Key Vault access logs must be enabled and monitored for suspicious activities, with alerts configured for any unauthorized attempts.
+
+#### Secure Application Integration
+
+Applications must access secrets via Managed Identities, and all hardcoded secrets must be removed from codebases.
+
+#### Compliance
+
+All Key Vault configurations and data handling processes must align with applicable UK government standards and data residency requirements.
+
+#### Exceptions
+
+Any deviation from this policy requires a documented approval from the Information Security Officer and an approved risk assessment.
+
+## Deployment
+
+### Code Scanning
+
+Before any code is pushed into an environment it should undergo our SAST checks from [Snyk](https://app.snyk.io/login) to try and ensure that the code we are creating is not going to leave us vulnerable. Any issues found that are high or above should be dealt with before doing any deployments to an environment. Where you feel there may be an exception to this, you can reach out to our [ITSO team](mailto:ukho-itso@ukho.gov.uk) or your Lead Developers for advice on mitigation/suppression of these issues.
+
+### Container Scanning
+
+Container images being produced for deployment into an environment should be scanned by a preferred scanning solution to ensure that we are not deploying vulnerable software. Details on our container policies can be found in the [Containers area.](/software-engineering-policies/Containers/) 
```

Comment:
Do we want to mention that container images created should be part of the ukho docker hub account? https://hub.docker.com/u/ukhydrographicoffice is that part of policy at all?